idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,321 @@
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            self.mscontactor.feed_side.component_list,
+            doc="permeance calculation",
+        )
+        def permeance_calculation(self, t, e, a, b):
+            return (
+                self.permeance[t, e, a] * self.selectivity[t, e, a, b]
+                == self.permeance[t, e, b]
+            )
+
+        p_units = feed_side_units.PRESSURE
+
+        crossover_component_list = list(
+            set(self.mscontactor.feed_side.component_list)
+            & set(self.mscontactor.sweep_side.component_list)
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            crossover_component_list,
+            doc="permeability calculation",
+        )
+        def permeability_calculation(self, t, s, m):
+            feed_side_state = self.mscontactor.feed_side[t, s]
+            if feed_side_state.get_material_flow_basis() is MaterialFlowBasis.molar:
+                mb_units = feed_side_units.FLOW_MOLE
+                rho = self.mscontactor.feed_side[t, s].dens_mol
+            elif feed_side_state.get_material_flow_basis() is MaterialFlowBasis.mass:
+                mb_units = feed_side_units.FLOW_MASS
+                rho = self.mscontactor.feed_side[t, s].dens_mass
+            else:
+                raise TypeError(
+                    "This model only supports MaterialFlowBasis equal to molar or mass"
+                )
+
+            return self.mscontactor.material_transfer_term[
+                t, s, "feed_side", "sweep_side", m
+            ] == -units.convert(
+                (
+                    rho
+                    * self.gpu_factor
+                    * self.permeance[t, s, m]
+                    * self.cell_area
+                    * (
+                        self.mscontactor.feed_side[t, s].pressure
+                        * self.mscontactor.feed_side[t, s].mole_frac_comp[m]
+                        - units.convert(
+                            self.mscontactor.sweep_side[t, s].pressure, to_units=p_units
+                        )
+                        * self.mscontactor.sweep_side[t, s].mole_frac_comp[m]
+                    )
+                ),
+                to_units=mb_units,
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            doc="Energy balance",
+        )
+        def energy_transfer(self, t, s):
+            return (
+                self.mscontactor.feed_side[t, s].temperature
+                == self.mscontactor.sweep_side[t, s].temperature
+            )
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {
+                "Feed Inlet": self.feed_side_inlet,
+                "Feed Outlet": self.feed_side_outlet,
+                "Permeate Inlet": self.sweep_side_inlet,
```

Comment:
You need logic here to handle cases where the sweep stream is not present.

Reply:
added